### PR TITLE
authorization_with_cancancan

### PR DESCRIPTION
**Pull request by sujoydatta**

### CODE REVIEW

**Comment by SSalekin** [comment]
ok.

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -1,16 +1,14 @@
 class Admin::CategoriesController < ApplicationController
-  before_action :load_category, only: [:destroy]
+  load_and_authorize_resource param_method: :category_params
```
Comment (dieunb) [inline app/controllers/admin/categories_controller.rb]: in short
```
load_and_authorize_resource
```

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
```
Comment (dieunb) [inline app/models/ability.rb]: remove `()`

File: app/models/ability.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    user ||= User.new
+    if user.admin
```
Comment (dieunb) [inline app/models/ability.rb]: `user.admin?`

File: app/models/user.rb [hunk 4]
```diff
@@ -1,7 +1,11 @@
 class User < ActiveRecord::Base
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
+  after_initialize :init
```
Comment (dieunb) [inline app/models/user.rb]: bad naming, should `init_admin`

File: app/models/user.rb [hunk 5]
```diff
@@ -1,7 +1,11 @@
...
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
   has_many :lessons
+
+  def init
+    self.admin = false if self.admin.nil?
```
Comment (dieunb) [inline app/models/user.rb]: in short
`admin = false if admin.nil?`

File: app/models/user.rb [hunk 6]
```diff
@@ -1,7 +1,11 @@
 class User < ActiveRecord::Base
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
+  after_initialize :init
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
   has_many :lessons
+
+  def init
```
Comment (dieunb) [inline app/models/user.rb]: private function

File: app/models/user.rb (hunk 5 above)
Comment (sujoydatta) [inline app/models/user.rb]: doesn't working if i remove self...

**Pull request merged**